### PR TITLE
Optimize Task.guarantee

### PR DESCRIPTION
**Pull request by alexandru**
Currently `guarantee` is based on `bracket`, but its implementation could be simplified and made slightly more efficient.

This is something similar to what I did for `cats.effect.IO`, but I forgot about it.

### CODE REVIEW

File: monix-eval/shared/src/test/scala/monix/eval/TaskGatherUnorderedSuite.scala [hunk 1]
```diff
@@ -134,12 +134,24 @@ object TaskGatherUnorderedSuite extends BaseTestSuite {
...
     var errorsThrow = 0
-    val task1 = Task.raiseError[Int](ex).executeAsync.doOnFinish { x =>
-      if (x.isDefined) errorsThrow += 1; Task.unit
-    }
-    val task2 = Task.raiseError[Int](ex).executeAsync.doOnFinish { x =>
-      if (x.isDefined) errorsThrow += 1; Task.unit
-    }
+
+    val task1 = Task
+      .raiseError[Int](ex)
```
Comment (oleg-py) [inline monix-eval/shared/src/test/scala/monix/eval/TaskGatherUnorderedSuite.scala]: Is it scalafmt style? That separation of companion object and a factory method on different lines kinda bothers me

Comment (alexandru) [reply monix-eval/shared/src/test/scala/monix/eval/TaskGatherUnorderedSuite.scala]: Unfortunately yes. Not sure what to do about it.

Comment (lorandszakacs) [reply monix-eval/shared/src/test/scala/monix/eval/TaskGatherUnorderedSuite.scala]: Huh, that's weird. I think you can do:
```
optIn {
  breaksInsideChains = false
  breakChainOnFirstMethodDot = false //pretty sure it's this one causing this.
}
```

File: monix-eval/shared/src/test/scala/monix/eval/TaskBracketSuite.scala [hunk 2]
```diff
@@ -152,34 +152,68 @@ object TaskBracketSuite extends BaseTestSuite {
     }
   }
 
-  test("bracket is stack safe") { implicit sc =>
+  test("bracket works with auto-cancelable run-loops") { implicit sc =>
```
Comment (Avasil) [inline monix-eval/shared/src/test/scala/monix/eval/TaskBracketSuite.scala]: Would it be possible to have a test that `acquire` is uncancelable?

**Pull request merged**